File: CPP/PegawaiRuko.cpp
#pragma once
#include <iostream>
#include <string>
#include <list>
#include "Orang.cpp"
#include "Ruko.cpp"

using namespace std;


class PegawaiRuko : public Orang
{
private:
    string jabatan;
    int gaji;
    string tanggalMulaiKerja;
    Ruko tempatBekerja;
public:
    PegawaiRuko()
    {


    }


    PegawaiRuko(string nama, int usia, string alamat, string noTelp, string jabatan, int gaji, string tanggalMulaiKerja, Ruko tempatBekerja) : Orang(nama, usia, alamat, noTelp)
    {
        this->jabatan = jabatan;
        this->gaji = gaji;
        this->tanggalMulaiKerja = tanggalMulaiKerja;
        this->tempatBekerja = tempatBekerja;
    }

    // Getter methods
    string getJabatan() {
        return this->jabatan;
    }

    int getGaji() {
        return this->gaji;
    }

    string getTanggalMulaiKerja() {
        return this->tanggalMulaiKerja;
    }

    Ruko getTempatBekerja() {
        return this->tempatBekerja;
    }

    // Setter methods
    void setJabatan(string jabatan) {
        this->jabatan = jabatan;
    }

    void setGaji(int gaji) {
        this->gaji = gaji;
    }

    void setTanggalMulaiKerja(string tanggalMulaiKerja) {
        this->tanggalMulaiKerja = tanggalMulaiKerja;
    }

    void addTempatKerja(Ruko tempatBekerja) {
        this->tempatBekerja = tempatBekerja;
    }
    
    ~PegawaiRuko()
    {


    }
};

File: PY/PegawaiRuko.py
from Orang import Orang
from Ruko import Ruko

class PegawaiRuko(Orang):
    __jabatan = ""
    __gaji = 0
    __tanggalMulaiKerja = ""
    __tempatBekerja = Ruko()
    
    def __init__(self, nama="", usia=0, alamat="", noTelp="", jabatan="", gaji=0, tanggalMulaiKerja="", tempatBekerja=Ruko()):
        super().__init__(nama, usia, alamat, noTelp)
        self.__jabatan = jabatan
        self.__gaji = gaji
        self.__tanggalMulaiKerja = tanggalMulaiKerja
        self.__tempatBekerja = tempatBekerja
    
    # Setter methods
    def setJabatan(self, jabatan):
        self.__jabatan = jabatan
    
    def setGaji(self, gaji):
        self.__gaji = gaji
    
    def setTanggalMulaiKerja(self, tanggalMulaiKerja):
        self.__tanggalMulaiKerja = tanggalMulaiKerja
    
    def addTempatKerja(self, tempatBekerja):
        self.__tempatBekerja = tempatBekerja
    
    # Getter methods
    def getJabatan(self):
        return self.__jabatan
    
    def getGaji(self):
        return self.__gaji
    
    def getTanggalMulaiKerja(self):
        return self.__tanggalMulaiKerja
    
    def getTempatKerja(self):
        return self.__tempatBekerja
